File: internal/parser/testdata/Sample.java
import java.util.List;
import java.util.Map;
import java.io.*;

/**
 * Represents a user in the system.
 * Handles user data and operations.
 */
public class User extends BaseEntity implements Serializable, Comparable<User> {

    private String name;
    private String email;
    protected int age;

    /**
     * Creates a new User with the given name and email.
     * @param name the user's name
     * @param email the user's email address
     */
    public User(String name, String email) {
        this.name = name;
        this.email = email;
    }

    /**
     * Returns the user's display name.
     * @return the formatted display name
     */
    public String getDisplayName() {
        return name + " <" + email + ">";
    }

    private void validateEmail(String email) {
        if (email == null || !email.contains("@")) {
            throw new IllegalArgumentException("Invalid email");
        }
    }

    protected List<String> getRoles() {
        return List.of("user");
    }

    @Override
    public int compareTo(User other) {
        return this.name.compareTo(other.name);
    }

    /**
     * Inner class for user preferences.
     */
    public static class Preferences {
        private Map<String, String> settings;

        public String getSetting(String key) {
            return settings.get(key);
        }
    }
}

/**
 * Defines the repository interface.
 */
interface UserRepository {
    User findById(long id);
    List<User> findAll();
    void save(User user);
}

enum UserRole {
    ADMIN,
    EDITOR,
    VIEWER
}

class UserServiceTest {
    @Test
    public void testCreateUser() {
        User user = new User("Alice", "alice@test.com");
        assert user.getDisplayName().contains("Alice");
    }

    @Test
    void testValidation() {
        // test validation logic
    }
}

File: internal/parser/testdata/sample.kt
import kotlin.collections.List
import kotlin.io.println

/**
 * Represents a user in the system.
 * Handles user data and operations.
 */
open class User(
    val name: String,
    private val email: String,
    protected var age: Int
) : BaseEntity(), Serializable {

    /**
     * Returns the formatted display name.
     */
    fun getDisplayName(): String {
        return "$name <$email>"
    }

    private fun validateEmail(): Boolean {
        return email.contains("@")
    }

    protected fun getRoles(): List<String> {
        return listOf("user")
    }

    companion object {
        fun create(name: String): User {
            return User(name, "", 0)
        }
    }
}

/**
 * A data class for user preferences.
 */
data class UserPreferences(
    val theme: String,
    val language: String,
    val notifications: Boolean
)

/**
 * Repository interface for user operations.
 */
interface UserRepository {
    fun findById(id: Long): User?
    fun findAll(): List<User>
    fun save(user: User)
}

sealed class Result<out T> {
    data class Success<T>(val data: T) : Result<T>()
    data class Error(val message: String) : Result<Nothing>()
    object Loading : Result<Nothing>()
}

object AppConfig {
    val version = "1.0.0"
    fun getEnvironment(): String = "production"
}

/**
 * Processes input data and returns results.
 */
fun processData(input: String, count: Int): List<String> {
    return List(count) { input }
}

fun String.toSlug(): String {
    return this.lowercase().replace(" ", "-")
}

internal fun helperFunction(x: Int, y: Int): Int {
    return x + y
}

class UserTests {
    @Test
    fun testCreateUser() {
        val user = User("Alice", "alice@test.com", 30)
        assert(user.name == "Alice")
    }

    @Test
    fun testDisplayName() {
        val user = User("Bob", "bob@test.com", 25)
        assert(user.getDisplayName().contains("Bob"))
    }
}
